pages/index.js

```diff
@@ -1,17 +1,38 @@
-import React from "react";
+import React, { useState } from "react";
 import Head from "next/head";
 
 import ContactList from "../components/ContactList";
+import FavoriteList from "../components/FavoriteList";
 
 export default function Home() {
+  
+  const [favorites, setFavorites] = useState([])
+
+  const addFavorites = (favorite) => {
+    if (!favorites.some(alreadyFavorite => alreadyFavorite == favorite)) {
+      setFavorites([...favorites, favorite])
+    }
+  }
+  
+  const removeFavorites = firstName => {
+    setFavorites(favorites.filter(item => item !== firstName))
+  }
```

Comment:
I'd recommend utilizing redux for your list of favorites for a couple reasons:
1. you wouldn't have to drill down so far with your props (favorites, addFavorites, removeFavorites)
2. if the app were to expand to more pages in the future, you wouldn't lose your favorites list every time you navigate away from the page

Also, by only using the first name of each contact for your favorites list, you've introduced a bug where you can't add two contacts with the same first name to your favorites list (Ex. Richard)